### ISSUE TITLE
Migrating schema between cloud projects using REST endpoints is broken

### ISSUE THREAD

**Issue by jakecollins1** [directus/directus#17980]
### Describe the Bug

We are using a directus cloud project for our production site, and I am trying to bring up another cloud project to use for our staging site.  I am trying to migrate the schema from the prod project to the staging project using the rest endpoints `/schema/snapshot`, `/schema/diff`, and `/schema/apply`.  
On the /schema/apply step, I am getting a strange error: 
`message: '"diff.collections[1].diff[0].lhs" must be of type object'`.  
When I log out the diff object, it looks like this: 
`diff: [ { kind: 'E', path: [Array], lhs: 3, rhs: 7 } ]`

The diff that is output from a directus endpoint has a value of `3` for `lhs`, and then a different directus endpoint expects that to be an `object`.  This seems like it is clearly an issue on the directus side, especially since the production project was built through the cloud client.

### To Reproduce

Try to migrate the schema from one cloud project to another using the rest endpoints `/schema/snapshot`, `/schema/diff`, and `/schema/apply`.  Our schema includes multiple collections, with one to many relationships defined between several of them.

### Hosting Strategy

Directus Cloud

**Comment by sleithart** [comment]
this has been happening to me as well, but from local => staging

**Comment by LilaRest** [comment]
I'm also experiencing this, even in local setup.

**Comment by MrSunshyne** [comment]
Wait, you can use migration endpoints on the cloud projects? I have a community cloud project and it's stuck on the version 9.21.2. From what I understand Schema Migration Endpoints are exposed in 9.23.0, and since there's only two weeks left before the community cloud gets shut down, i'm not sure how I can migrate my projects to a self hosted instance. 

[I've posted on the discord](https://discord.com/channels/725371605378924594/1072936534719873055), but the "cloud" team seems to be different from the dev team. Your issues indicates that not all cloud instances are running on the same version, which means I might get stuck on the old version till they shut it down..

I also reached out to the cloud team and had a call with them last week, but no solution from them since then.

I would really appreciate a way to at least use the schema migration endpoints so that I can safely move my data and schema to another instance. 

TLDR; The email to community cloud admins says use the Schema Migration endpoints, but they're not available on my instance.

**Comment by paescuj** [comment]
Closed via #18048
